force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,45 @@
+public with sharing class OpportunityTriggerHandler {
+    public static void amountValidation(List<Opportunity> oppoNewList){
+        List<Opportunity> oppoListForUpdate = new List<Opportunity>();
+        for (Opportunity oppo : oppoNewList) {
+            if (oppo.Amount < 5000) {
+                oppo.Amount.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+    public static void deleteCloseWonOpportunity(List<Opportunity> oppoDelList){
+        List<Opportunity> preventOppoList = new List<Opportunity>();
+
+        Map<Id, Opportunity> oppoMap = new Map<Id, Opportunity>();
+        for (Opportunity oppo : oppoDelList) {
+            oppoMap.put(oppo.AccountId, oppo);
+        }
+        Map<Id, String> accsMap = new Map<Id, String>();
+        List<Account> accList = [SELECT Industry FROM Account WHERE Id IN : oppoMap.keySet()];
+        for (Account acc : accList) {
+            accsMap.put(acc.Id, acc.Industry);
+        }
+        for (Opportunity oppo : oppoMap.values()) {
```

Comment:
Try to combine these two maps into something like this
Map<Id, Account> accountsMap = new Map<Id, Account>(
			[SELECT Id, Industry FROM Account WHERE Id IN (SELECT AccountId FROM Opportunity WHERE Id IN :oppoDelList)]
		);